bundles/org.openhab.core.automation/src/test/java/org/openhab/core/automation/internal/module/handler/ItemStateConditionHandlerTest.java

```diff
@@ -65,9 +65,9 @@ public ParameterSet(String comparisonState, State itemState, boolean expectedRes
                 { new ParameterSet("5", new DecimalType(23), false) }, //
                 { new ParameterSet("5", new DecimalType(5), true) }, //
                 { new ParameterSet("5 °C", new DecimalType(23), false) }, //
-                { new ParameterSet("5 °C", new DecimalType(5), false) }, //
+                { new ParameterSet("5 °C", new DecimalType(5), true) }, //
```

Comment:
I would claim that this change is not correct.
The unit test was just right before: 5 is NOT the same as 5 °C as it is lacking any unit and could thus well also be 5 °F.

Reply:
Yes, I agree.

Reply:
Keeping this in mind we should adopt the remaining comparisons too and add a conversion to a base unit before. I will have a look.